tests/console/check_package_version.pm

```diff
@@ -0,0 +1,75 @@
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Check package version only for sles12sp5 migration scenarios
+# Maintainer: Yutao Wang <yuwang@suse.com>
+
+use strict;
+use warnings;
+use version;
+use base 'installbasetest';
+use testapi;
+use utils 'systemctl', 'zypper_call';
+
+my @packages = qw(autofs openscap augeas freeradius-server gpgme libgpg-error0); #apache2-mod_auth_openidc);
+my @packagev = qw(5.1.5 1.3.0 1.0.0 3.0.18 1.9.0 1.17);# 0.0.0);
+
+my @packages_s390x = qw(libnuma-devel);
+my @packagev_s390x = qw(2.0.12);
+
+sub search_packages {
+    my ($package) = @_;
+    zypper_call("se -s $package");
+}
+
+sub cmp_version {
+    my ($old, $new) = @_;
+    my @newv = split(/-/, $new);
+    my $v1 = version->parse($old);
+    my $v2 = version->parse($newv[0]);
+    return $v1 <= $v2;
+}
+
+sub cmp_result {
+    my ($pcks, $pckv) = @_;
+    search_packages($pcks);
```

Comment:
what the point of this line ? you not checking anything you just running `zypper se` with same list of packages just before doing the same again in next line but this time with some actual checks

Reply:
I just confirm the package can be found. If it not existed, it will return error. At the openQA, we can get the red square.

Reply:
if package will be not found you get a failure in next lines anyway so you increasing test execution time without good reason

Reply:
I removed this line. :)

Reply:
I believe you ... but still can you please push your changes to github so I can really see it ? :)